lib/geo-index.js

```diff
@@ -131,37 +131,39 @@ module.exports = function (tzGeojson, dataDir, targetIndexPercent, callback) {
     }
   }
 
-  var i, j
+  const unindexableDataIndex = {}
+  let filePosition = 0
 
   // analyze each unindexable area in a queue, otherwise the program may run out
   // of memory
-  var unindexableAreaAnalyzingQueue = async.queue(
-    function (unindexableData, cb) {
-      var features = []
-      // calculate intersected area for each intersected zone
-      for (j = unindexableData.intersectedZones.length - 1; j >= 0; j--) {
-        var tzIdx = unindexableData.intersectedZones[j]
-        var intersectedGeoJson = getIntersectingGeojson(
-          tzIdx,
-          unindexableData.curBoundsGeometry
-        )
+  function writeUnindexableData(unindexableData, outfile) {
+    var features = []
+    // calculate intersected area for each intersected zone
+    for (let j = unindexableData.intersectedZones.length - 1; j >= 0; j--) {
+      var tzIdx = unindexableData.intersectedZones[j]
+      var intersectedGeoJson = getIntersectingGeojson(
+        tzIdx,
+        unindexableData.curBoundsGeometry
+      )
 
-        if (intersectedGeoJson) {
-          intersectedGeoJson.properties.tzid = data.timezones[tzIdx]
-          features.push(intersectedGeoJson)
-        }
+      if (intersectedGeoJson) {
+        intersectedGeoJson.properties.tzid = data.timezones[tzIdx]
+        features.push(intersectedGeoJson)
       }
+    }
 
-      var areaGeoJson = featureCollection(features)
-      var path = dataDir + '/' + unindexableData.curZone.id.replace(/\./g, '/')
+    var areaGeoJson = featureCollection(features)
 
-      fileWritingQueue.push(
-        { folder: path, filename: 'geo.buf', data: areaGeoJson },
-        cb
-      )
-    },
-    10
-  )
+    const buf = Buffer.from(geobuf.encode(areaGeoJson, new Pbf()))
+    outfile.write(buf)
+
+    unindexableDataIndex[unindexableData.curZone.id] = {
```

Comment:
unindexableData.curZone.id.split('.').join('') or use regex to eliminate all the .s to improve lookup efficiency